File: CSharp/Data_Structures/Hashtables/exercices.cs
namespace CSharp.Data_Structures.Hashtables;

public class exercices
{
    static int FirstRecurringCharacter(int[] tab)
    {
        Dictionary<int, int> myMap = new Dictionary<int, int>();
        int index = 0;
        foreach (int val in tab)
        {
            if (myMap.ContainsKey(val))
            {
                return val;
            }
            myMap.Add(val, index);
        }

        return -1;
    }
    
    

    public static void main()
    {
        Console.WriteLine(FirstRecurringCharacter(new[]{1,2,3,2}));
    }
}

File: Python/Data_Structures/Hashtables/exercices.py
def first_recurring_character(tab: list) -> int:
    my_map = {}
    for index, value in enumerate(tab):
        if value in my_map.keys():
            return value
        my_map[value] = index
    return -1


if __name__ == '__main__':
    print(first_recurring_character([1, 2, 3]))
